ros/api/v1/hosts.py

```diff
@@ -204,6 +204,9 @@ def build_system_filters():
                 else:
                     abort(400, message='Not a valid RHEL version')
             filters.append(System.operating_system.in_(modified_operating_systems))
+        if group_names := request.args.getlist('group_name'):
+            query = System.groups[0]['name'].astext.in_(group_names)
+            filters.append(query)
```

Comment:
```suggestion
            filters.append(System.groups[0]['name'].astext.in_(groups))
```

Reply:
This won't check for all groups,
~~~
(Pdb) query = db.session.query(System.groups[0])
(Pdb) result = db.session.execute(query)
(Pdb) result.all()
[({'id': '7cbdaad6-d97e-4ed0-9095-ef07d73b4839', 'name': 'ros-test-1'},)]
(Pdb) query = db.session.query(System.groups[1])
(Pdb) result = db.session.execute(query)
(Pdb) result.all()
[({'id': '99999999-d97e-4ed0-9095-ef07d73b4839', 'name': 'ros-test-2'},)]
~~~

Reply:
@r14chandra for code consistency I updated the code by removing the query variable and pushed the changes in the changes in the latest commit. Have a look and let me know! 🙌🏻

Reply:
@upadhyeammit Sorry I didn't get you, do you mean the query won't return all the systems with that `group_name`?

Reply:
> @upadhyeammit Sorry I didn't get you, do you mean the query won't return all the systems with that `group_name`?

Actually one system can be in multiple groups and when you execute groups[0] it only checks with first element of array. The check should be for array of groups and not for just first element. Unfortunately I dont have answer for you on this one, as I am still playing with the queries.

Reply:
> Actually one system can be in multiple groups

Is it mentioned somewhere?

Reply:
@upadhyeammit yes thats correct but for the MVP (Which we are focusing at the moment) the system can be part of only one group for now and hence I wrote the query like this. I feel let's move ahead if it looks good for now and when inventory will start supporting multiple groups feature we will also adapt to that?
[Here is the document link](https://docs.google.com/document/d/1_zWWRpunieVbvuR6SMjP2fReA1YOS1XruLQaTnYjb6A/edit#heading=h.rjwpur2zwfgz) where I read about the single group scope, have a look and let me know your thoughts 🙌🏻

Reply:
Wow! thats good to know, i have been battling to accommodate the multiple groups! Yeah then your changes makes sense!

Reply:
great, thanks for clarifying it out @upadhyeammit and let me know if any more questions :)